tests/index.test.js

```diff
@@ -0,0 +1,23 @@
+import { map, filter } from '../index.js';
+
+describe('map', () => {
+  it('replaces .map with a .length function', () => {
+    const numbers = [1, 2, 3, 4, 5];
+    const modification = (item) => item * 2;
+    const actual = map(numbers, modification);
+
+    expect(actual).toEqual([2, 4, 6, 8, 10]);
+    expect(numbers).toEqual([1, 2, 3, 4, 5]);
```

Comment:
I love that you're testing to make sure you didn't mutate the original,  this is really useful to monitor as you work.